HW1/P3/P3.py

```diff
@@ -0,0 +1,19 @@
+import pyspark
+
+# initialize spark context
+sc = pyspark.SparkContext("local[4]", "Spark1")
+
+# create RDD of word list from text file
+wlist = sc.textFile('EOWL_words.txt')
+
+# associate sorted character list & counter with each word
+slist = wlist.map(lambda w: (''.join(sorted(w)), (1, [w])))
+
+# combine words with same sorted character list
+anagrams = slist.reduceByKey(lambda (n1, wlst1), (n2, wlst2): (n1 + n2, wlst1 +wlst2))
+
+# flatten (since reduceByKey wouldn't play nice with multiple values)
+anagrams_flat = anagrams.map(lambda (a, (b,c)): (a,b,c))
```

Comment:
I think this has to do with the representation of key value pairs in an RDD.

With a 3-tuple, you don't have a clear key-value relationship, while it's clearly evident with a 2-tuple.